compiler/rustc_middle/src/lint.rs

```diff
@@ -389,7 +390,7 @@ pub fn struct_lint_level(
             // if this lint occurs in the expansion of a macro from an external crate,
             // allow individual lints to opt-out from being reported.
             let not_future_incompatible =
-                future_incompatible.map(|f| f.reason.edition().is_some()).unwrap_or(true);
+                future_incompatible.is_none_or(|f| f.reason.edition().is_some());
```

Comment:
The `not_` prefix of the name here also hints at an inverted condition that might be easier to read:

```rust
let incompatible = future_incompatible.is_some_and(|f| f.reason.edition().is_none());
```